Shopify/cli
# Add jsx-transform support to extensions build

### WHY are these changes introduced?

React 17 has support for jsx without explicit React import, we want to have the same experience when building checkout extensions


### WHAT is this pull request doing?

Added `automatic` option for `esbuild` options, to allow auto import of React
More details: https://esbuild.github.io/api/#jsx
### How to test your changes?

- Create an extension
- choose `typescript-react` template
- Observe `tsconfig.json` file is generated
- Make sure react is not imported in your component files
- After building/deploying make sure you don't see `React not defined` error in your console
<img width="459" alt="Screenshot 2022-10-05 at 13 36 15" src="https://user-images.githubusercontent.com/4366612/194055654-ca3a7977-dac5-428b-8b56-9fcc5a26bd90.png">

### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
